packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1854,7 +2231,132 @@ class _TrackTapGestureRecognizer extends TapGestureRecognizer {
+      return false;
+    }
+    final CustomPaint customPaint = customPaintKey.currentContext!.widget as CustomPaint;
+    final ScrollbarPainter painter = customPaint.foregroundPainter! as ScrollbarPainter;
+    final Offset localOffset = _getLocalOffset(customPaintKey, offset);
+    return painter.hitTestButtonsInteractive(localOffset, kind, isLeadingButton: _isLeadingButton);
+  }
+}
+
+// A long press gesture detector that responds to events on the leading scrollbar's button.
+class _LeadingButtonPressGestureRecognizer extends _ButtonPressGestureRecognizer {
+  _LeadingButtonPressGestureRecognizer({
+    double? postAcceptSlopTolerance,
+    Set<PointerDeviceKind>? supportedDevices,
+    required Object debugOwner,
+    required GlobalKey customPaintKey,
+    required Duration pressDuration,
+  }) : super(
+    isLeadingButton: true,
+    postAcceptSlopTolerance: postAcceptSlopTolerance,
+    supportedDevices: supportedDevices,
+    debugOwner: debugOwner,
+    customPaintKey: customPaintKey,
+    pressDuration: pressDuration,
+  );
+}
+
+// A long press gesture detector that responds to events on the trailing scrollbar's button.
+class _TrailingButtonPressGestureRecognizer extends _ButtonPressGestureRecognizer {
+  _TrailingButtonPressGestureRecognizer({
+    double? postAcceptSlopTolerance,
+    Set<PointerDeviceKind>? supportedDevices,
+    required Object debugOwner,
+    required GlobalKey customPaintKey,
+    required Duration pressDuration,
+  }) : super(
+    isLeadingButton: false,
+    postAcceptSlopTolerance: postAcceptSlopTolerance,
+    supportedDevices: supportedDevices,
+    debugOwner: debugOwner,
+    customPaintKey: customPaintKey,
+    pressDuration: pressDuration,
+  );
+}
+
 Offset _getLocalOffset(GlobalKey scrollbarPainterKey, Offset position) {
   final RenderBox renderBox = scrollbarPainterKey.currentContext!.findRenderObject()! as RenderBox;
   return renderBox.globalToLocal(position);
 }
+
+/// Scrollbar button states that can take on when receiving input from the user.
+enum ScrollbarButtonState {
+  /// The state when the user drags their mouse cursor over the button.
+  hovered,
+
+  /// The state when the user is actively pressing down on the button.
+  pressed,
+}
+
+/// The presentation style of the scrollbar button.
+enum ScrollbarButtonStyle {
+  /// No buttons displayed.
+  none,
+
+  /// Two buttons are displayed split at both ends of the track.
+  split,
+
+  /// Two buttons are displayed together on the leading of the track.
+  groupedLeading,
+
+  /// Two buttons are displayed together on the trailing of the track.
+  groupedTrailing,
+}
+
+/// Base class for Scrollbar leading and trailing button shapes.
+///
+abstract class ScrollbarButtonShape {
+  /// This abstract const constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const ScrollbarButtonShape();
+
+  /// Return the button size.
+  double getButtonSize() => 18.0;
```

Comment:
We might want this to be meeting guidelines by default (48), and then when enabling it for Windows, we can set the size to match the native button in that case.